scipy/optimize/_linprog.py

```diff
@@ -135,742 +141,1009 @@ def linprog_terse_callback(xk, **kwargs):
+    b : 1-D array
+        1-D array of values representing the RHS of each equality constraint
+        (row) in A (for standard form problem).
+    c : 1-D array
+        Coefficients of the linear objective function to be minimized (for
+        standard form problem).
+    c0 : float
+        Constant term in objective function due to fixed (and eliminated)
+        variables.
 
-        x : ndarray
-            The independent variable vector which optimizes the linear
-            programming problem.
-        fun : float
-            Value of the objective function.
-        slack : ndarray
-            The values of the slack variables.  Each slack variable corresponds
-            to an inequality constraint.  If the slack is zero, then the
-            corresponding constraint is active.
-        success : bool
-            Returns True if the algorithm succeeded in finding an optimal
-            solution.
-        status : int
-            An integer representing the exit status of the optimization::
+    References
+    ----------
+    .. [6] Bertsimas, Dimitris, and J. Tsitsiklis. "Introduction to linear
+           programming." Athena Scientific 1 (1997): 997.
 
-             0 : Optimization terminated successfully
-             1 : Iteration limit reached
-             2 : Problem appears to be infeasible
-             3 : Problem appears to be unbounded
+    """
 
-        nit : int
-            The number of iterations performed.
-        message : str
-            A string descriptor of the exit status of the optimization.
+    if sps.issparse(A_eq):
+        sparse = True
+        A_eq = sps.lil_matrix(A_eq)
+        A_ub = sps.lil_matrix(A_ub)
 
-    Examples
-    --------
-    Consider the following problem:
+        def hstack(blocks):
+            return sps.hstack(blocks, format="lil")
 
-    Minimize: f = -1*x[0] + 4*x[1]
+        def vstack(blocks):
+            return sps.vstack(blocks, format="lil")
 
-    Subject to: -3*x[0] + 1*x[1] <= 6
-                 1*x[0] + 2*x[1] <= 4
-                            x[1] >= -3
+        zeros = sps.lil_matrix
+        eye = sps.eye
+    else:
+        sparse = False
+        hstack = np.hstack
+        vstack = np.vstack
+        zeros = np.zeros
+        eye = np.eye
+
+    fixed_x = set()
+    if len(undo) > 0:
+        # these are indices of variables removed from the problem
+        # however, their bounds are still part of the bounds list
+        fixed_x = set(undo[0])
+    # they are needed elsewhere, but not here
+    bounds = [bounds[i] for i in range(len(bounds)) if i not in fixed_x]
+    # in retrospect, the standard form of bounds should have been an n x 2
+    # array. maybe change it someday.
+
+    # modify problem such that all variables have only non-negativity bounds
+
+    bounds = np.array(bounds)
+    lbs = bounds[:, 0]
+    ubs = bounds[:, 1]
+    m_ub, n_ub = A_ub.shape
+
+    lb_none = np.equal(lbs, None)
```

Comment:
Same with the previous case we don't need any conversion to `dtype=object` arrays. Lists are much faster and convenient.